pkg/cloud/instance.go

```diff
@@ -159,6 +167,22 @@ func (c *client) GetOrCreateVMInstance(
 
 	if len(csMachine.Spec.AffinityGroupIds) > 0 {
 		p.SetAffinitygroupids(csMachine.Spec.AffinityGroupIds)
+	} else if !(strings.ToLower(csMachine.Spec.Affinity) == "no" || csMachine.Spec.Affinity == "") {
```

Comment:
While logically correct, I glossed-over the leading ! while reading, and then wondered why you were creating affinity groups when the param is "no" or blank.  Took me a head-scratch or two to find the !.  For readability I prefer:

if (strings.ToLower(csMachine.Spec.Affinity) != "no" && csMachine.Spec.Affinity != "")

It's reads closer to how I'd say it too: if it's not no and it's not blank.

Reply:
Yup, fixed. Thanks!